frontend/src/component/demo/demo-topics.tsx

```diff
@@ -0,0 +1,489 @@
+                        Edit the existing variants.
+                    </Typography>
+                ),
+            },
+            {
+                target: 'button[data-testid="MODAL_ADD_VARIANT_BUTTON"]',
+                content: (
+                    <Typography variant="body2" color="text.secondary">
+                        Add a new variant to the list.
+                    </Typography>
+                ),
+                backCloseModal: true,
+            },
+            {
+                target: 'div[data-testid="variant_list"] > div:last-of-type input',
+                content: (
+                    <>
+                        <Typography variant="body2" color="text.secondary">
+                            Enter a new variant name.
+                        </Typography>
+                        <Typography
+                            variant="body2"
+                            color="text.secondary"
+                            sx={{ mt: 1 }}
+                        >
+                            We recommend choosing a{' '}
+                            <a
+                                href="https://developer.mozilla.org/en-US/docs/Web/CSS/named-color"
+                                target="_blank"
+                            >
+                                color
+                            </a>
+                            .
+                        </Typography>
+                        <Typography variant="body2" color="text.secondary">
+                            Example: <Badge as="span">aqua</Badge>.
+                        </Typography>
+                    </>
+                ),
+                backCloseModal: true,
+                nextButton: true,
+            },
+            {
+                target: 'div[data-testid="variant_list"] > div:last-of-type input[name="variant-payload-value"]',
+                content: (
+                    <Typography variant="body2" color="text.secondary">
+                        Enter the{' '}
+                        <a
+                            href="https://developer.mozilla.org/en-US/docs/Web/CSS/named-color"
+                            target="_blank"
+                        >
+                            color
+                        </a>{' '}
+                        you chose on the previous step as the payload.
+                    </Typography>
+                ),
+                nextButton: true,
+            },
+            {
+                target: 'div[data-testid="variant_list"] > div:last-of-type > div:last-of-type > button',
+                content: (
+                    <Typography variant="body2" color="text.secondary">
+                        Let's also add an override for our user.
+                    </Typography>
+                ),
+            },
+            {
+                target: 'div[data-testid="variant_list"] > div:last-of-type div[data-testid="context_field"]',
```

Comment:
These are sensitive to refactors and we don't have a good testing strategy to validate these continue to work. Can we use a more direct way of identifying the targets? What worries me is the `> div: last-of-type`. 

Something like: `'div[data-testid="variant_list"] div[data-testid="context_field"]:last'` indicating the last context_field inside variant_list would be better IMO. But it also means that these `data-testid`'s are not unique (they are repeated, right?).

Alternatively, if we could **identify** the last context_field in variants' list, something like `[data-testid="variants_list_last_context_field"]`, this would make it less error prone, but it requires some additional code to only add this to the last element we want to target. But at the same time this can open up the possibility of adding tests to the page to guarantee that the element exists.

Reply:
Yep, had the same thought, I think these are valid concerns. I think I'll have another look through the steps and try to use more explicit selectors whenever possible.

However, if you think about it, the example of `> div:last-of-type` may not be so bad. We're looking at the `variants-list` and selecting the last element of the list (the last variant we just added). I don't see another trivial way of doing this that would be much more explicit...

I also prefer to adapt the demo guide to the product whenever possible, instead of the other way around.

Worst case scenario, if we eventually end up with scenarios where the user gets stuck in one of the steps, they can always escape from it.

Reply:
Tried addressing this with more explicit selectors in https://github.com/Unleash/unleash/pull/3569/commits/6f1258b337b43a45cd717ef766da492e29ee8730 but it's something to keep in mind for the future.